cmd/orch-tester/orch_tester.go

```diff
@@ -717,13 +717,18 @@ func (s *streamerClient) sampleFromPlaylist(mid string) (source, rendition []byt
 
 	r := rand.Int()
 	segIdx := r % int(math.Min(float64(renditionPl.Len()), float64(sourcePl.Len())))
-	seqNo := sourcePl.Segments[segIdx].SeqId // Assumes that segIdx points to the same seqNo in the slice of sources and renditions
-	rendition, err = s.downloadSegment(fmt.Sprintf("%v/%v", s.broadcaster, renditionPl.Segments[seqNo].URI))
+	seqNo := sourcePl.Segments[segIdx].SeqId
+	rendURI := renditionPl.Segments[seqNo].URI
+	urlSplit := strings.Split(rendURI, "/")
+	fName := urlSplit[len(urlSplit)-1]
+
+	rendition, err = s.downloadSegment(fmt.Sprintf("%v/%v", s.broadcaster, rendURI))
 	if err != nil {
 		return nil, nil, "", err
 	}
 
-	source, err = s.downloadSegment(fmt.Sprintf("%v/%v", s.broadcaster, sourcePl.Segments[seqNo].URI))
+	sourceURI := fmt.Sprintf("%v/stream/%v_0_0/source/%v", s.broadcaster, mid, fName)
```

Comment:
This URL structure doesn't seem right. Looks like `s.broadcaster` would be included twice in the end URL passed to `downloadSegment()`.